Update all dependencies

WhoamiHF opened the pull request:
- compilation ok
- login page loading ok
- workspaces page loading ok
- css not applied

TODO:
- [x] [Bootstrap](https://getbootstrap.com/docs/5.3/migration/): Dropped form-specific layout classes for our grid system. Use our grid and utilities instead of .form-group, .form-row, or .form-inline. -> replace `form-group` with `mb-3`
	- [x] We can possibly use `mb-3` (margin bottom) from Bootstrap instead of custom `withPadding`
- [x] When there is an error in the panel, e.g., when I put a numeric signal instead of datetime in the linechart panel as time signal and the Elasticsearch queries fail, the panel keeps reloading over and over again. Previously, the plot just displayed "Error loading data." and the panel did not reload.
	- Apparently, the `setFlashMessage` method refreshes the page 
- [x] In LineChart, shift+mouse wheel does not work (it should zoom the chart)... check the `createChartZoom()` method in `TimeBasedChartBase`
- [x] Histogram and Scatterplot don't work, but it is hard to debug with the panel reloading mentioned above. Ask Michal for more details after the reloading is resolved.
- [x] When I'm not logged in, and I open directly `http://localhost:8080/workspaces/sample2`, it should redirect me to the login page (`http://localhost:8080/login?next=%2Fworkspaces%2Fsample2`). Right now, it just crashes.
- [x] Figure out how to make the sidebar collapsable on mobile (see TODO in code in `page.js`)
- [x] Workspace name is missing in the breadcrumbs
- [x] When a panel is open, the sidebar should show other panels in the same workspace.
- [x] (@Mnaukal): check built-in templates
- [x] (@Mnaukal): check embedding
- [x] (@Mnaukal): check PDF export (see https://github.com/smartarch/ivis-core/wiki/PDF-export if it doesn't work)
- [x] check user creation
- [x] update `SubRoute` for the new router
- [x] update navigation blocking when form is not saved for the new router
	- This is probably not worth fixing now. It will require a major refactoring of the `Section`, `SectionContent` and `RenderRoute` components in `page.js`, because of changes in react-router@v6 (the `useBlocker` hook requires [data API router](https://reactrouter.com/en/main/routers/picking-a-router) which has a different schema for specifying the routes).
	- There might be a workaround or a library that I'm not aware of.
- [ ] remove the debugging `console.log`s

This is not a new error (it was in the older version), but I just wanted to note it somewhere:

- [ ] I think there is something slightly wrong with the permissions. When I open a panel, I should be able to edit its parameters... there should be an "Edit settings" button in the panel's settings (the gear icon in the top right-hand corner of the panel). However, the button only appears if the panel's template has "Elevated access" (and I think it should appear always as the "admin" should have the "edit" permission for the panel).

WhoamiHF committed:
- problem with mysql
- updated dependencies
- fixed compilation warnings and errors on client

WhoamiHF committed:
- fixed problem with mysql
- fixed errors on the logging page in the browser and
- fixed logout
- fixed languages

Mnaukal commented:
Check the browser developer tools to see whether the URL to the CSS files is correct (you don't want to see `Failed to load resource: the server responded with a status of 404 (NOT FOUND)`). I think the problem might again be in the configuration, you might need to copy parts of the `www` configuration section from `production.yaml` to `developement.yaml` (check if you have HTTPS enabled or not, it might be necessary to update `trustedUrlBase` to only have `http://` without `s`).

If this doesn't help, I will try to install it on my machine and check.

Btw, there is no need to close and create a new pull request, you can just push to the `extremeXP` branch and the code of the PR updates.

WhoamiHF committed:
- problem with language and account dropper
- problem with settings sidebar

Mnaukal committed:
Small fix in installation

Mnaukal committed:
Fix dropdown

Mnaukal committed:
Remove CoreUI

Mnaukal committed:
Add Sidebar

Mnaukal committed:
Fix Buffer (base64-url)

Mnaukal committed:
Update alert close button

WhoamiHF committed:
- fixed Uncaught TypeError: jqModal.modal is not a function
- dubious ownership is because of chown in functions

WhoamiHF committed:
- fixed sample 2
- dealing with cors

WhoamiHF committed:
- updated elastic search
- little bit improved styles
- fixed edit shown two times (and other problems with menu)
- removed types because of elastic search versions
- repaired bugs with modals (not possible to close, setting state of unmounted..)

In process:
Validation of any form fails when the values were not changed

WhoamiHF committed:
- problem with fetching data for custom panels

Mnaukal committed:
Update modal buttons

Commit:
Installable

Commit:
Installable - replaced absolute path used for testing with relative

Commit:
Installable - fixed loading of scss

Mnaukal commented on client/package.json:
```diff
@@ -8,86 +8,92 @@
+    "css-loader": "^5.2.7",
+    "file-loader": "^6.2.0",
+    "nodemon": "^3.0.3",
+    "node-sass": "^6.0.1",
```
Prosím nepřidávat `node-sass`, je deprecated. Místo toho bych upravil wrapper, aby používal `sass` (stejně jako `"css-compile"` výše v tomto souboru).

Mnaukal commented on client/src/lib/i18n.js:
```diff
@@ -60,11 +60,12 @@ export default i18n;
 export const TranslationContext = React.createContext(null);
 
-export const withTranslation = createComponentMixin({
+export const withTranslationCustom = createComponentMixin({
```
Nešlo by to udělat tak, aby se nezměnil ten export a nebylo potřeba upravovat importy ve skoro všech souborech? Aby v tomhle souboru nevznikal konflikt jmen, tak stačí udělat `import {I18nextProvider, withTranslation as BLABLABLA} from 'react-i18next';`. A všude jinde chceme importovat jen ten náš `withTranslation` a ne ten z `'react-i18next'`.

Mnaukal commented on client/src/lib/modals.js:
```diff
@@ -239,33 +251,28 @@ export function tableAddDeleteButton(actions, owner, perms, deleteUrl, name, del
     }
 
     if (!perms || perms.includes('delete')) {
-        if (owner.tableRestActionDialogData.shown) {
```
Tenhle `if` je smazaný úmyslně?

Mnaukal commented on server/index.js:
```diff
@@ -1,5 +1,15 @@
+const cors = require('cors');
+const app = express();
+
+app.use(cors({
```
Ty "cors" by neměly být potřeba, ne? To jsme myslím vyřešili bez nich.

Mnaukal commented on server/lib/indexers/elasticsearch-query.js:
```diff
@@ -498,15 +498,15 @@ class QueryProcessor {
                     elsAgg.histogram = {
                         ...this.getField(field),
-                        interval: agg.computedStep || 1e-16 /* FIXME - this is  a hack, find better way to handle situations when there is no interval */,
+                        fixed_interval: agg.computedStep || 1e-16 /* FIXME - this is  a hack, find better way to handle situations when there is no interval */,
```
Myslím, že tady mělo zůstat jen "`interval"`, ten `"fixed_interval"` se týká jen `date_histogram`.

[dokumentace histogram](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-bucket-histogram-aggregation.html)  
[dokumentace date histogram](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-bucket-datehistogram-aggregation.html)

Mnaukal commented on server/lib/router-async.js:
```diff
@@ -16,6 +17,14 @@ function replaceLastBySafeHandler(handlers) {
 function create() {
     const router = new express.Router();
 
+    const cors = require('cors');
```
Tady taky zůstalo "cors".

Mnaukal commented on server/routes/index.js:
```diff
@@ -37,7 +37,6 @@ function getRouter(appType) {
                 csrfToken: req.csrfToken(),
                 ivisConfig: JSON.stringify(ivisConfig),
                 indexFile: appType === AppType.TRUSTED ? getTrustedUrl('client/index-trusted.js') : getSandboxUrl('client/index-sandbox.js'),
-                bodyClass: appType === AppType.TRUSTED ? "trusted" : "sandbox",
```
Tohle si myslím, že chceme zachovat. "trusted" sice možná žádné styly nemá, ale "sandbox" určitě ano (viz řádek 26 v `ivis.scss`)

Mnaukal commented on server/views/layout.hbs:
```diff
@@ -10,9 +10,8 @@
     <script src="{{publicPath}}static-npm/bootstrap.min.js"></script>
-    <script src="{{publicPath}}static-npm/coreui.min.js"></script>
 </head>
-<body class="ivis {{bodyClass}}">
```
Tohle si myslím, že chceme zachovat (rozlišení "trusted" a "sandbox").

Mnaukal reviewed (commented):
Ahoj Davide, zběžně jsem prošel tvoje změny a narazil na pár drobností, kde si myslím, že by to chtělo trošku upravit. Někde jsi smazal části, které si myslím, že by měly zůstat zachované, ale to se vůbec nedivím, vzhledem k velikosti projektu vůbec je těžké všimnout si všeho. Díky.

WhoamiHF committed:
- removed node-sass
- renamed withTranslationCustom to withTranslation, resolved conflict
- removed cors artifacts
- returned trusted/sandbox tag

todo
- if in modals
- interval x fixed_interval in elasticsearch-query

WhoamiHF committed:
- removed node-sass
- renamed withTranslationCustom to withTranslation, resolved conflict
- removed cors artifacts
- returned trusted/sandbox tag
- return if in modals
- changed fixed_interval to interval for number data in elasticsearch-query

Mnaukal committed:
Remove checkcss

Mnaukal committed:
Fix modals.js (revert unnecessary changes)

Mnaukal commented on client/src/lib/page-common.js:
```diff
@@ -84,10 +86,17 @@ async function resolve(route, match, prevResolverState) {
+                            .then(response => {
+                                return response;
+                            })
+                            .catch(error => {
```
Je tenhle `catch` k něčemu potřeba? Já bych čekal, že když tu nebyl, tak se ty výjimky zachytily někde výše. Ono to totiž způsobuje ten problém se špatným redirectem při přihlášení, protože to tady chytí `NotLoggedInError`, což by nemělo. Jestli není důvod tu ten `catch` přidávat, tak bych vrátil předchozí verzi.

Mnaukal commented on client/src/workspaces/panels/WorkspacePanel.js:
```diff
@@ -59,11 +60,11 @@ export default class WorkspacePanel extends Component {
 
     componentDidMount() {
-        extractPermanentLinkAndRedirect(this.props.location, this.props.history);
+        extractPermanentLinkAndRedirect(this.props.location, this.props.navigate);
```
Tady asi není správně upravený vnitřek té funkce, uvnitř je pořád `history`.

Mnaukal committed:
Update sidebar styles

Mnaukal commented on client/src/root-trusted.js:
```diff
@@ -84,37 +84,86 @@ import {TranslationRoot} from "./lib/i18n";
-        structure: (resolved, params) => {
+        panelRender: props => <WorkspacePanel panel={props.resolved.panel}/>
+
+        /*structure: (resolved, params) => {
```
Tohle taky zůstalo zakomentované... ale přiznám se, že nemám tušení, co to má dělat, protože mám pocit, že zobrazování panelů funguje i bez toho. Tak to zatím asi můžeme nechat být, jen jsem to tu chtěl označit.

@bures, ty tušíš, k čemu tenhle kus struktury je? Na mě to na první pohled působí jako nějaké další podstránky pod panelem, ale to jsem nikdy neviděl použité, tak právě nevím, jestli to potřebujeme.

Mnaukal commented on client/src/root-trusted.js:
```diff
@@ -231,42 +278,71 @@ const getStructure = t => {
                                 link: params => `/settings/workspaces/${params.workspaceId}/edit`,
                                 navs: {
-                                    ':action(edit|delete)': {
+                                    /*':action(edit|delete)': {
```
Tady je to níže rozepsané, takže to zakomentované se asi může dát pryč?

Mnaukal commented on client/src/lib/page.js:
```diff
@@ -332,11 +336,13 @@ class PanelRoute extends Component {
     }
 }
 
+const DummyComponent = () => {
```
Tohle bylo asi jen pro debugování a může se to smazat.

Mnaukal commented on client/src/lib/page-common.js:
```diff
@@ -366,53 +380,71 @@ class SubRoute extends Component {
-                    return <Route key={route.path} exact={route.exact} path={route.path} render={render} />
-                };
-
+                const childRoute = routes[0];
```
Tohle taky vypadá nedodělaně, ale asi to nemá smysl teď řešit, protože `SubRoute` se použije, jen když existuje `route.structure`, což se týká jen panelů a je to zatím zakomentované (a nevím, jestli se to někde doopravdy používá).

Mnaukal committed:
Hopefully fixed `setFlashMessage` reloading panels

Mnaukal commented on client/src/lib/page.js:
```diff
@@ -418,13 +423,14 @@ export class SectionContent extends Component {
     componentDidMount() {
-        window.addEventListener('beforeunload', this.beforeUnloadHandler);
-        this.historyUnblock = this.props.history.block('Changes you made may not be saved. Are you sure you want to leave this page?');
+        /*window.addEventListener('beforeunload', this.beforeUnloadHandler);
```
Tohle bychom asi chtěli taky zprovoznit, ať to uživatele varuje při opuštění neuloženého formuláře.

Mnaukal commented on client/src/lib/page.js:
```diff
@@ -368,26 +374,25 @@ export class BeforeUnloadListeners {
         };
 
-        this.historyUnlisten = props.history.listen((location, action) => {
+        /*this.historyUnlisten = props.history.listen((location, action) => {
```
A tohle se hodí na automatické smazání "flashMessage", taky by to chtělo odkomentovat a upravit pro nový router.

Mnaukal committed:
Second attempt to fix `setFlashMessage` reloading panels

Mnaukal committed:
Fix breadcrumbs and sidebar for panels

Mnaukal committed:
Fix zoom in line chart

Mnaukal committed:
Fix bugs in ScatterPlotBase.js

Mnaukal reviewed (commented):
Ahoj Davide, povedlo se mi projít další část tvých změn. Narazil jsem tam na pár věcí, které ještě nejsou dodělané (zůstaly zakomentované). Většina z toho se týká routeru, tak jsem tě chtěl poprosit, jestli bys na to ještě mrknul, přeci jenom máš s tou novou verzí routeru trochu víc zkušeností, když už jsi nějaké úpravy dělal. Ale pokud vůbec netušíš, co s tím, tak to můžeme prostě sepsat do toho TODO v popisu pull requestu a pak nějak postupně vyřešit (já jsem mezitím některé z těch TODO vyřešil). Tak jen dej vědět, jestli se do těch dodělávek budeš pouštět nebo ne (třeba jako odpovědi na ty příslušné komentáře). Každopádně prioritu má tvorba docker kontejneru pro instalaci a základní verze API pro ExtremeXP, tohle klidně nech až potom. Díky.

Mnaukal committed:
Small style fixes

Mnaukal committed:
Collapse sidebar on narrow screens

Mnaukal committed:
Fix recursion error (brush+zoom) in HistogramChart.js

Mnaukal committed:
Allow CTRL+S to save template/task code

Mnaukal committed:
Add example application for embedding

Commit:
- Working querying executed-workflows with characteristics

- Added scripts setup/create_indexes.sh and executed-workflows-test.sh

Mnaukal committed:
Revert "- Working querying executed-workflows with characteristics"

This reverts commit 99674ba497ca9aaf209109926a836ef0f09c9ebc.

Mnaukal committed:
Fix login redirect

Mnaukal committed:
Cleanup

Mnaukal committed:
Add source maps to simplify debugging

Mnaukal committed:
Fix hiding flash message on page change

Mnaukal committed:
Fix permanent links

Mnaukal committed:
Mark TODOs in the code

Mnaukal committed:
Fix bug in job creation (when selecting default built-in task)

Mnaukal committed:
Add example of builtin template, fix SubRoute

Mnaukal committed:
Fix title in example builtin template

Mnaukal committed:
Fix navigation blocking when form is not saved

Mnaukal committed:
Clean up unnecessary code after navigation blocking fix (36e96c1)

Mnaukal committed:
Remove debugging `console.log`s

Mnaukal committed:
Clean up imports

Mnaukal committed:
Remove resolved TODO

Mnaukal committed:
Set min-height of top menu

Mnaukal committed:
Create .gitattributes to set the correct line endings for Shell scripts

Mnaukal committed:
Add basic Docker support to ivis-core

The Docker files are a modified version from the ExpVis wrapper.
